src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/MissingGeneratorAttributeRuleTests.cs

```diff
@@ -0,0 +1,406 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.SourceGeneratorAttributeAnalyzer,
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.SourceGeneratorAttributeAnalyzerFix>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.SourceGeneratorAttributeAnalyzer,
+    Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.SourceGeneratorAttributeAnalyzerFix>;
+
+namespace Microsoft.CodeAnalysis.Analyzers.UnitTests.MetaAnalyzers
+{
+    public class MissingGeneratorAttributeRuleTests
+    {
+        private const string SourceGeneratorStub_CSharp = @"
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.CodeAnalysis
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class GeneratorAttribute : Attribute
+    {
+    }
+
+    public interface ISourceGenerator
+    {
+        void Initialize(GeneratorInitializationContext context);
+        void Execute(GeneratorExecutionContext context);
+    }
+
+    public struct GeneratorInitializationContext
+    {
+    }
+
+    public readonly struct GeneratorExecutionContext
+    {
+    }
+}";
+
+        private const string SourceGeneratorStub_VisualBasic = @"
+Imports System
+Imports Microsoft.CodeAnalysis
+
+Namespace Microsoft.CodeAnalysis
+    <AttributeUsage(AttributeTargets.Class)>
+    public Class GeneratorAttribute
+            Inherits Attribute
+    End Class
+
+    public interface ISourceGenerator
+        Sub Initialize(context As GeneratorInitializationContext)
+        Sub Execute(context As GeneratorExecutionContext)
+    End Interface
+
+    public Class GeneratorInitializationContext
+    End Class
+
+    public Class GeneratorExecutionContext
+    End Class
+End Namespace";
```

Comment:
❗ Do not attempt to reinvent a public API in tests. Rather, use the `ReferenceAssemblies` property to add a reference to  version of the library which contains these definitions (in this case, Microsoft.CodeAnalysis.Common 3.10.0). This ensures the analyzer works as expected in production scenarios (we've had bugs in the test definition manifest in broken production features with no test failures).

Reply:
What's the correct way to reference a specific min version of an assembly? I thought we were restricted by what assemblies the tests reference, which is currently 3.3 and doesn't have ISourceGenerator etc.

Reply:
`ReferenceAssemblies.Default.AddPackages`

Reply:
That's awesome, I didn't realize it would work with NuGet packages!!! Fixing